experimental/Schemes/SpecOpen.jl

```diff
@@ -630,9 +656,31 @@ the numerator and denominator of ``f`` have to be elements of
 the ring ``𝕜[x₁,…,xₙ]``.
 """
 function maximal_extension(
-    X::AbsSpec{<:Ring, <:AbsLocalizedRing}, 
+    X::AbsSpec{<:Ring, <:MPolyLocalizedRing}, 
+    f::AbstractAlgebra.Generic.Frac{RET}
+  ) where {RET<:MPolyElem}
+
+  a = numerator(f)
+  b = denominator(f)
+  g = gcd(a, b)
+  if !isone(g)
+    a = divexact(a, g)
+    b = divexact(b, g)
+    f = parent(f)(a,b)
+  end
+  W = OO(X)
+  I = ideal(W, b)
+  U = SpecOpen(X, I)
```

Comment:
No, you want SpecOpen(X,[b]) as you want the complement of the vanishing locus of b.